firedrake/mesh.py

```diff
@@ -2601,6 +2677,85 @@ def _on_rank_vertices(vertex_coords, parent_mesh):
     return np.compress(on_rank, vertex_coords, axis=0)
 
 
+def RelabeledMesh(mesh, indicator_functions, subdomain_ids, **kwargs):
+    """Construct a new mesh that has new subdomain ids.
+
+    :arg mesh: base :class:`~.MeshGeometry` object using which the
+        new one is constructed.
+    :arg indicator_functions: list of indicator functions that mark
+        selected entities (cells or facets) as 1; must use
+        "DP"/"DQ" (degree 0) functions to mark cell entities and
+        "HDiv Trace" (degree 0) functions to mark facet entities.
+    :arg subdomain_ids: list of subdomain ids associated with
+        the indicator functions in indicator_functions; thus,
+        must have the same length as indicator_functions.
+    :kwarg name: optional name of the output mesh object.
+    """
+    import firedrake.function as function
+
+    if not isinstance(mesh, MeshGeometry):
+        raise TypeError(f"mesh must be a MeshGeometry, not a {type(mesh)}")
+    tmesh = mesh.topology
+    if isinstance(tmesh, VertexOnlyMeshTopology):
+        raise NotImplementedError("Currently does not work with VertexOnlyMesh")
+    elif isinstance(tmesh, ExtrudedMeshTopology):
+        raise NotImplementedError("Currently does not work with ExtrudedMesh; use RelabeledMesh() on the base mesh and then extrude")
+    if not isinstance(indicator_functions, (list, tuple)) or \
+       not isinstance(subdomain_ids, (list, tuple)):
```

Comment:
`isinstance(..., collections.abc.Sequence)`

Reply:
check for integer labels (`isinstance(..., numbers.Integral)`)

Reply:
Fixed.